### PR TITLE
fix batch_norm parameter share

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/3984

### CODE REVIEW

File: python/paddle/trainer_config_helpers/layers.py [hunk 1]
```diff
@@ -2957,6 +2957,7 @@ def batch_norm_layer(input,
                      bias_attr=None,
                      param_attr=None,
                      layer_attr=None,
+                     mean_var_names=None,
```
Comment (qingqing01) [inline python/paddle/trainer_config_helpers/layers.py]: Put the new argument as the last one.

File: python/paddle/trainer_config_helpers/layers.py [hunk 2]
```diff
@@ -3014,6 +3015,8 @@ def batch_norm_layer(input,
     :type param_attr: ParameterAttribute
     :param layer_attr: Extra Layer Attribute.
     :type layer_attr: ExtraLayerAttribute
+    :param mean_var_names: [mean name, variance name]
+    :type mean_var_names: string list
```
Comment (qingqing01) [inline python/paddle/trainer_config_helpers/layers.py]: Put the new argument as the last one.

File: python/paddle/trainer/config_parser.py [hunk 3]
```diff
@@ -2366,6 +2366,7 @@ def __init__(self,
                  inputs,
                  bias=True,
                  use_global_stats=True,
+                 mean_var_names=None,
```
Comment (qingqing01) [inline python/paddle/trainer/config_parser.py]:
```python
# ...
moving_average_fraction=0.9,		              
batch_norm_type=None,
mean_var_names=None,
**xargs):		            
```

Comment (chengduoZH) [reply python/paddle/trainer/config_parser.py]: Done

File: python/paddle/trainer_config_helpers/layers.py (hunk 1 above)
Comment (chengduoZH) [reply python/paddle/trainer_config_helpers/layers.py]: Done

File: python/paddle/trainer_config_helpers/layers.py (hunk 2 above)
Comment (chengduoZH) [reply python/paddle/trainer_config_helpers/layers.py]: Done

**Pull request merged**